pkg/server/api/bundle/v1/service_test.go

```diff
@@ -1079,6 +1079,242 @@ func TestBatchCreateFederatedBundle(t *testing.T) {
 	}
 }
 
+func TestBatchUpdateFederatedBundle(t *testing.T) {
+	test := setupServiceTest(t)
+	defer test.Cleanup()
+
+	_, expectedX509Err := x509.ParseCertificates([]byte("malformed"))
+	require.Error(t, expectedX509Err)
+
+	for _, tt := range []struct {
+		name              string
+		bundlesToUpdate   []*types.Bundle
+		preExistentBundle *common.Bundle
+		inputMask         *types.BundleMask
+		outputMask        *types.BundleMask
+		expectedResults   []*bundlepb.BatchCreateFederatedBundleResponse_Result
+		expectedLogMsgs   []spiretest.LogEntry
+		dsError           error
+	}{
+		{
+			name:              "Update succeeds with nil masks",
+			preExistentBundle: &common.Bundle{TrustDomainId: federatedTrustDomain.IDString()},
+			bundlesToUpdate: []*types.Bundle{
+				makeValidBundle(t, federatedTrustDomain),
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{
+					Status: api.CreateStatus(codes.OK, `bundle updated successfully for trust domain: "another-example.org"`),
+					Bundle: makeValidBundle(t, federatedTrustDomain),
+				},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.InfoLevel,
+					Message: `Bundle updated successfully`,
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+					},
+				},
+			},
+		},
+		{
+			name:              "Only values set in input mask are updated",
+			preExistentBundle: &common.Bundle{TrustDomainId: federatedTrustDomain.IDString()},
+			bundlesToUpdate:   []*types.Bundle{makeValidBundle(t, federatedTrustDomain)},
+			inputMask: &types.BundleMask{
+				RefreshHint: true,
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{
+					Status: api.CreateStatus(codes.OK, `bundle updated successfully for trust domain: "another-example.org"`),
+					Bundle: &types.Bundle{
+						TrustDomain: federatedTrustDomain.String(),
+						RefreshHint: makeValidBundle(t, federatedTrustDomain).RefreshHint,
+					},
+				},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.InfoLevel,
+					Message: `Bundle updated successfully`,
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+					},
+				},
+			},
+		},
+		{
+			name:              "Only values set in output mask are included in the response",
+			preExistentBundle: &common.Bundle{TrustDomainId: federatedTrustDomain.IDString()},
```

Comment:
looks like in almost all cases you have a preexistentbundle, maybe you can do it part of the test body? and not a parameter?

Reply:
Hmm I rather keep it as a parameter. It is only used on 3 of 9 tests and I believe it will be required for tests that have a pre-existent invalid bundle.